gax/src/main/java/com/google/api/gax/batching/v2/BatcherImpl.java

```diff
@@ -167,18 +175,28 @@ public void close() throws InterruptedException {
     private final RequestBuilder<ElementT, RequestT> builder;
     private final List<SettableApiFuture<ElementResultT>> results;
     private final BatchingDescriptor<ElementT, ElementResultT, RequestT, ResponseT> descriptor;
+    private final long elementThreshold;
+    private final long bytesThreshold;
+
+    private long elementCounter = 0;
+    private long byteCounter = 0;
 
     private Batch(
         RequestT prototype,
-        BatchingDescriptor<ElementT, ElementResultT, RequestT, ResponseT> descriptor) {
+        BatchingDescriptor<ElementT, ElementResultT, RequestT, ResponseT> descriptor,
+        BatchingSettings batchingSettings) {
       this.descriptor = descriptor;
       this.builder = descriptor.newRequestBuilder(prototype);
+      this.elementThreshold = batchingSettings.getElementCountThreshold();
+      this.bytesThreshold = batchingSettings.getRequestByteThreshold();
       this.results = new ArrayList<>();
     }
 
     void add(ElementT element, SettableApiFuture<ElementResultT> result) {
```

Comment:
I think that in a future PR, it would be nice for this method to create its own SettableApiFuture and return an ApiFuture

Reply:
Thanks for the suggestion, I would address this change in the next PR.